src/librustc_metadata/creader.rs

```diff
@@ -260,13 +239,26 @@ impl<'a> CrateLoader<'a> {
             .map(|trait_impls| (trait_impls.trait_id, trait_impls.impls))
             .collect();
 
+        let def_path_table = record_time(&self.sess.perf_stats.decode_def_path_tables_time, || {
+            crate_root.def_path_table.decode((&metadata, self.sess))
+        });
+
+        let raw_proc_macros =  crate_root.proc_macro_data.map(|_| {
+            if self.sess.opts.debugging_opts.dual_proc_macros {
+                let host_lib = host_lib.as_ref().unwrap();
+                self.dlsym_proc_macros(host_lib.dylib.as_ref().map(|p| p.0.clone()),
+                                       &host_lib.metadata.get_root(), span)
+            } else {
+                self.dlsym_proc_macros(dylib.clone().map(|p| p.0), &crate_root, span)
+            }
+        });
```

Comment:
Nit: could you move this code to its old place (it's hard to see the difference otherwise).